burn-core/src/optim/adamw.rs

```diff
@@ -0,0 +1,371 @@
+                    0.057032, -0.036518, -0.382951, 0.232516, 0.173738, -0.309182,
+                ],
+                [
+                    -0.038703, 0.016052, -0.313155, 0.225982, -0.295039, 0.289981,
+                ],
+                [
+                    -0.314920, -0.237394, -0.387704, -0.315067, -0.095153, 0.141081,
+                ],
+                [
+                    0.306815, -0.234226, 0.348083, -0.191115, 0.356002, -0.049993,
+                ],
+                [-0.035634, -0.030083, 0.104636, 0.170244, 0.009196, 0.359580],
+            ]),
+            Data::from([
+                -0.406555, 0.067568, -0.115982, 0.096477, 0.115287, -0.007080,
+            ]),
+        );
+
+        let t_state_updated: Tensor<TestADBackend, 2> =
+            Tensor::from_data(state_updated.weight.to_data());
+        let t_state_expected: Tensor<TestADBackend, 2> =
+            Tensor::from_data(state_expected.weight.to_data());
+
+        let t_actual_difference = t_state_updated.sub(t_state_expected);
+        let expected_difference: Tensor<TestADBackend, 2> = Tensor::from_floats([
+            [
+                -0.016695, -0.019573, -0.023942, -0.023132, -0.020668, -0.020566,
+            ],
+            [
+                -0.020668, -0.018018, -0.016251, -0.022484, -0.021762, -0.016982,
+            ],
+            [
+                -0.019703, -0.018548, -0.016955, -0.022418, -0.017039, -0.023019,
+            ],
+            [
+                -0.016920, -0.015994, -0.016204, -0.016967, -0.019053, -0.021519,
+            ],
+            [
+                -0.023185, -0.016026, -0.023617, -0.018215, -0.023598, -0.019593,
+            ],
+            [
+                -0.019734, -0.018083, -0.021164, -0.021856, -0.020104, -0.023720,
+            ],
+        ]);
+
+        t_actual_difference
+            .into_data()
+            .assert_approx_eq(&expected_difference.into_data(), ASSERT_PRECISION);
+
+        let (weight_updated, bias_updated) = (
+            state_updated.weight.to_data(),
+            state_updated.bias.unwrap().to_data(),
+        );
+        let (weight_expected, bias_expected) = (
+            state_expected.weight.to_data(),
+            state_expected.bias.unwrap().to_data(),
+        );
+
+        bias_updated.assert_approx_eq(&bias_expected, ASSERT_PRECISION);
+        weight_updated.assert_approx_eq(&weight_expected, ASSERT_PRECISION);
+    }
+
+    fn given_linear_layer(weight: Data<f32, 2>, bias: Data<f32, 1>) -> nn::Linear<TestADBackend> {
+        let linear = nn::LinearConfig::new(6, 6).init();
+        let record = given_linear_record(weight, bias);
+
+        linear.load_record(record)
+    }
+
+    fn given_linear_record(
+        weight: Data<f32, 2>,
+        bias: Data<f32, 1>,
+    ) -> nn::LinearRecord<TestADBackend> {
+        nn::LinearRecord {
+            weight: Param::from(Tensor::from_data(weight)),
+            bias: Some(Param::from(Tensor::from_data(bias))),
+        }
+    }
```

Comment:
I feel like this could be shorten.

```rust
fn given_linear_layer(weight: Data<f32, 2>, bias: Data<f32, 1>) -> nn::Linear<TestADBackend> {
    let record = nn::LinearRecord {
        weight: Param::from(Tensor::from_data(weight)),
        bias: Some(Param::from(Tensor::from_data(bias))),
    };

    nn::LinearConfig::new(6, 6).init_with(record)
}
```

Reply:
Agree, will push changes now